pkg/hive/cell/structured.go

```diff
@@ -0,0 +1,384 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/trigger"
+	"github.com/google/uuid"
+	"golang.org/x/exp/maps"
+)
+
+type children map[string]struct{}
+
+// GetSubScope creates a new context with a sub reporter scope context.
+// This creates a new node in the structured health reporter tree, and any calls
+// to GetReporter from the returned context will return a reporter that is a child
+// of this reporter tree.
+//
+// GetSubScope can be chained together to create various levels of sub reporters.
+//
+// Example:
+//
+// 1. Init root scope (note: this is provided to modules automatically).
+// root := rootScope(ctx, hr)
```

Comment:
Examples use `ctx` and the indentation for the examples seems to be a bit messed up.